File: src/ois_subjects/subjects.pas
{ NOTE: it is recommended to use this even if you don't understand the following code }

const
    MAXP = 1000000;
    MAXN = 100000;

var
    M, N, P, i, j : LongInt;
    K             : Array[0..MAXN-1] of LongInt;
    S             : Array[0..MAXN-1] of Array[0..5] of LongInt;
    ans           : Array[0..MAXP-1] of Array[0..1] of LongInt;

begin
{
    uncomment the two following lines if you want to read/write from files
    assign(input,  'input.txt');  reset(input);
    assign(output, 'output.txt'); rewrite(output);
}

    ReadLn(N, M);

    for j := 0 to N - 1 do 
    begin        
        Read(K[j]);
        for i := 0 to K[j] - 1 do
            Read(S[j][i]);
    end;


    P := 0;
    for i := 0 to 2 - 1 do
        for j := 0 to MAXP - 1 do
            ans[j][i] := 0;

    { INSERT YOUR CODE HERE }

    WriteLn(P);
    for j := 0 to P - 1 do 
    begin
        for i := 0 to 2 - 1 do
            Write(ans[j][i], ' ');
        WriteLn();
    end;

end.

File: src/ois_subjects/subjects.c
// NOTE: it is recommended to use this even if you don't understand the following code.

#include <assert.h>
#include <stdio.h>
#include <stdlib.h>
#include <stdbool.h>

#define MAXP 1000000
#define MAXN 100000

int M, N, P, i, j;
int K[MAXN];
int S[MAXN][5];
int ans[MAXP][2];

int main() {
    // uncomment the two following lines if you want to read/write from files
    // freopen("input.txt", "r", stdin);
    // freopen("output.txt", "w", stdout);

    assert(2 == scanf("%d%d", &N, &M));

    for (j = 0; j < N; ++j) {
        assert(1 == scanf("%d", &K[j]));
        for (i = 0; i < K[j]; ++i)
            assert(1 == scanf("%d", &S[j][i]));
    }

    P = 0;


    // INSERT YOUR CODE HERE


    printf("%d\n", P);
    for (j = 0; j < P; ++j) {
        for (i = 0; i < 2; ++i)
            printf("%d ", ans[j][i]);
        printf("\n");
    }

    return 0;
}
